specs/SharedBuffer.md

```diff
@@ -0,0 +1,333 @@
+}
+
+```
+
+## .NET
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    class CoreWebView2Environment
+    {
+        public CoreWebView2SharedBuffer CreateSharedBuffer(ulong size);
+    }
+
+    class CoreWebView2SharedBuffer : System.IDisposable
+    {
+        public ulong Size { get; };
+        
+        /// The raw memory address of the buffer.
+        /// You can cast it to pointer to real data types like byte* to access the memory
+        /// from `unsafe` code region.
+        /// Normal app should use `GetStream` to get a Stream object to access the buffer.
+        public IntPtr Buffer { get; };
+        
+        /// The native file mapping handle of the shared memory of the buffer.
+        /// Normal app should use `GetStream` to get a Stream object to access the buffer.
+        /// For advanced scenario, you could use native APIs to duplicate this handle to
+        /// another application process and create a mapping from the duplicated handle in
+        /// that process to access the buffer from that separate process.
+        public IntPtr Handle { get; };
+        
+        public Stream GetStream();
+        
+        void Close();
+    }
+    
+    runtimeclass CoreWebView2
+    {
+        public void PostSharedBufferToScript(
+            CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+    
+    class CoreWebView2Frame
+    {
+        public void PostSharedBufferToScript(
+            CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+}
+
+```
+## WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2Environment
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Environment11")]
+        {
+            CoreWebView2SharedBuffer CreateSharedBuffer(UInt64 size);
+        }
+    }
+
+    runtimeclass CoreWebView2SharedBuffer : Windows.Foundation.IClosable
+    {
+        UInt64 Size { get; };
+        
+        Windows.Storage.Streams.IRandomAccessStream GetStream();
+        
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2SharedBuffer_Manual")]
+        {
+            /// A reference to the underlying memory of the shared buffer.
+            /// You can get IMemoryBufferByteAccess from the object to access the memory as an array of bytes.
+            /// See [I](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.imemorybufferreference?view=winrt-22621)
+            /// for more details.
+            Windows.Foundation.IMemoryBufferReference Buffer { get; };
+        }
+        
+        // Note that we are not exposing Handle from WinRT API.
+    }
+    
+    runtimeclass CoreWebView2
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_14")]
+        {
+            void PostSharedBufferToScript(
+                CoreWebView2SharedBuffer sharedBuffer, Boolean isReadOnlyToScript, String additionalDataAsJson);
```

Comment:
Boolean parameter is always confusing. Without the comment, you just see `true`, and it's not clear if this means "Yes, it is read-only" or "Yes, it is writable."

So you have to comment it every time. But then you didn't save anything. May as well be
`COREWEBVIEW2_SHAREDBUFFERACCESS_READONLY` and `COREWEBVIEW2_SHAREDBUFFERACCESS_READWRITE`.

Reply:
Sounds good to me. Will change to enums.

Reply:
Please update